testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/requestdeserialization/AbstactUnmarshallingFilterTestCase.java

```diff
@@ -81,7 +79,7 @@ static Serializable getTemplatesImpl()  {
         // This call executes in the server in a deployment that has this class's module
         // configured as a dependency
         try {
-            return (Serializable) Class.forName("org.apache.xalan.xsltc.trax.TemplatesImpl").newInstance();
+            return (Serializable) Class.forName("com.sun.org.apache.xalan.internal.xsltc.trax.TemplatesImpl").newInstance();
```

Comment:
This fails in the security manager tests.

```
java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "accessClassInPackage.com.sun.org.apache.xalan.internal.xsltc.trax")" in code source "(vfs:/content/UnmarshallingFilterDisabledTestCase.jar <no signer certificates>)" of "ModuleClassLoader for Module "deployment.UnmarshallingFilterDisabledTestCase.jar" from Service Module Loader")
	at org.wildfly.security.elytron-base@2.0.0.Beta3//org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:309)
	at org.wildfly.security.elytron-base@2.0.0.Beta3//org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:201)
	at java.base/java.lang.SecurityManager.checkPackageAccess(SecurityManager.java:1238)
	at org.wildfly.security.elytron-base@2.0.0.Beta3//org.wildfly.security.manager.WildFlySecurityManager.checkPackageAccess(WildFlySecurityManager.java:505)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:174)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at org.jboss.modules.JDKModuleFinder$JDKModuleLoader.loadClassLocal(JDKModuleFinder.java:158)
	at org.jboss.modules.Module.loadModuleClass(Module.java:765)
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:192)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:315)
	at deployment.UnmarshallingFilterDisabledTestCase.jar//org.jboss.as.test.integration.ejb.remote.requestdeserialization.AbstactUnmarshallingFilterTestCase.getTemplatesImpl(AbstactUnmarshallingFilterTestCase.java:82)
	at deployment.UnmarshallingFilterDisabledTestCase.jar//org.jboss.as.test.integration.ejb.remote.requestdeserialization.UnmarshallingFilterDisabledTestCase.testBlocklistDisabled(UnmarshallingFilterDisabledTestCase.java:80)
	at deployment.UnmarshallingFilterDisabledTestCase.jar//org.jboss.as.test.integration.ejb.remote.requestdeserialization.UnmarshallingFilterDisabledTestCase.testHttp(UnmarshallingFilterDisabledTestCase.java:75)
```

Reply:
Unfortunately the WildFly Preview CI job that uses SE17 does not run this test, which means we have no coverage on SE 17. :(  But I'm pretty sure it will fail as the server JPMS settings do not allow reflective access to the java.xml module.

Reply:
I added a Linxu - SE 17 job and triggered a retest so we'll see if I'm right.

Reply:
You are right. There is exception about access the javax.xml module:
```
Caused by: java.lang.IllegalAccessException: class org.jboss.as.test.integration.ejb.remote.requestdeserialization.AbstactUnmarshallingFilterTestCase cannot access class com.sun.org.apache.xalan.internal.xsltc.trax.TemplatesImpl (in module java.xml) because module java.xml does not export com.sun.org.apache.xalan.internal.xsltc.trax to unnamed module @7e071932
```
I added a `--add-open` in parent pom to fix this failure.

Reply:
> This fails in the security manager tests.
> 
> ```
> java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "accessClassInPackage.com.sun.org.apache.xalan.internal.xsltc.trax")" in code source "(vfs:/content/UnmarshallingFilterDisabledTestCase.jar <no signer certificates>)" of "ModuleClassLoader for Module "deployment.UnmarshallingFilterDisabledTestCase.jar" from Service Module Loader")
> 	at org.wildfly.security.elytron-base@2.0.0.Beta3//org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:309)
> 	at org.wildfly.security.elytron-base@2.0.0.Beta3//org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:201)
> 	at java.base/java.lang.SecurityManager.checkPackageAccess(SecurityManager.java:1238)
> 	at org.wildfly.security.elytron-base@2.0.0.Beta3//org.wildfly.security.manager.WildFlySecurityManager.checkPackageAccess(WildFlySecurityManager.java:505)
> 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:174)
> 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
> 	at org.jboss.modules.JDKModuleFinder$JDKModuleLoader.loadClassLocal(JDKModuleFinder.java:158)
> 	at org.jboss.modules.Module.loadModuleClass(Module.java:765)
> 	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:192)
> 	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
> 	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
> 	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
> 	at java.base/java.lang.Class.forName0(Native Method)
> 	at java.base/java.lang.Class.forName(Class.java:315)
> 	at deployment.UnmarshallingFilterDisabledTestCase.jar//org.jboss.as.test.integration.ejb.remote.requestdeserialization.AbstactUnmarshallingFilterTestCase.getTemplatesImpl(AbstactUnmarshallingFilterTestCase.java:82)
> 	at deployment.UnmarshallingFilterDisabledTestCase.jar//org.jboss.as.test.integration.ejb.remote.requestdeserialization.UnmarshallingFilterDisabledTestCase.testBlocklistDisabled(UnmarshallingFilterDisabledTestCase.java:80)
> 	at deployment.UnmarshallingFilterDisabledTestCase.jar//org.jboss.as.test.integration.ejb.remote.requestdeserialization.UnmarshallingFilterDisabledTestCase.testHttp(UnmarshallingFilterDisabledTestCase.java:75)
> ```

Fixed.